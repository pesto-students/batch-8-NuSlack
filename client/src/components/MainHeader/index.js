import React from 'react';
import {
  Icon, Row, Col, Avatar, Layout,
} from 'antd';
import styled from 'styled-components';
import CircleButton from '../CircleButton';
import SmallButton from '../SmallButton';
import { useHomeContext } from '../../context/HomeContext';

const { Header } = Layout;

const LoggedInUser = styled.div`
  padding: 10px;
  color: white;
  line-height: 2em;
  h1.user-name {
    color: white;
    font-size: 1.8em;
    margin: 0;
  }
`;

const UserStatus = styled.div`
  font-size: 1.4em;
  line-height: 1.5em;
`;

const HomeHeader = styled(Header)`
  height: 100px;
  background: #607d8b;
  @media only screen and (max-width: 600px) {
    padding: 0 25px;
    font-size: 11px;
    .ant-avatar {
      height: 60px !important;
      width: 60px !important;
    }
  }
`;

const MainHeader = () => {
  const { user, logoutUser } = useHomeContext();

  return (
    <HomeHeader className="home-header">
      <Row>
        <Col span={12}>
          <Row type="flex" align="middle" justify="start" style={{ height: '100px' }}>
            <Avatar shape="square" size={90} icon="user" />
            <LoggedInUser>
              <h1 className="user-name">{user && user.username}</h1>
              <UserStatus>
                <Icon type="check-circle" theme="twoTone" twoToneColor="#52c41a" />
                Online
              </UserStatus>
            </LoggedInUser>
          </Row>
        </Col>
        <Col span={12}>
          <Row type="flex" align="middle" justify="end" style={{ height: '100px' }}>
            <Col span={32}>
              <CircleButton>
                <Icon type="setting" />
              </CircleButton>
              &nbsp;
              <SmallButton onClick={logoutUser}>
                <Icon type="logout" /> Logout
              </SmallButton>
            </Col>
          </Row>
        </Col>
      </Row>
    </HomeHeader>
  );
};

export default MainHeader;
