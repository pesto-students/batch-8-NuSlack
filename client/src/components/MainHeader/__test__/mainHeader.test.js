import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import MainHeader from '../index';

describe('<MainHeader />', () => {
  it('should match the snapshot', () => {
    const tree = shallow(<MainHeader />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
