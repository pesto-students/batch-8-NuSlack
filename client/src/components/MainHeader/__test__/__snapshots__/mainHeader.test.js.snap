// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainHeader /> should match the snapshot 1`] = `
<Styled(Adapter)
  className="home-header"
>
  <Row
    gutter={0}
  >
    <Col
      span={12}
    >
      <Row
        align="middle"
        gutter={0}
        justify="start"
        style={
          Object {
            "height": "100px",
          }
        }
        type="flex"
      >
        <Avatar
          icon="user"
          shape="square"
          size={90}
        />
        <styled.div>
          <Link
            to="/profile"
          >
            <h1
              className="user-name"
            />
          </Link>
          <styled.div>
             Team: 
          </styled.div>
        </styled.div>
      </Row>
    </Col>
    <Col
      span={12}
    >
      <Row
        align="middle"
        gutter={0}
        justify="end"
        style={
          Object {
            "height": "100px",
          }
        }
        type="flex"
      >
        <Link
          to="/teams"
        >
          <Button
            block={false}
            ghost={false}
            htmlType="button"
            loading={false}
            type="primary"
          >
            Invitations
          </Button>
        </Link>
         
        <TeamsDrawer />
         
        <SmallButton>
          <Icon
            type="logout"
          />
           Logout
        </SmallButton>
      </Row>
    </Col>
  </Row>
</Styled(Adapter)>
`;
